### ISSUE TITLE
Color differences between CPU and GPU decoded frames

### ISSUE THREAD

**Issue by NicolasHug** [pytorch/torchcodec#425]
From @pjs102793's  https://github.com/pytorch/torchcodec/issues/424#issuecomment-2522514388

to generate sample video: `ffmpeg -f lavfi -i testsrc2=size=1920x1080:rate=30 -t 1 testsrc2.mp4`

On main [b0de666](https://github.com/pytorch/torchcodec/commit/b0de66677bac322e628f04ec90ddeeb0304c6abb)

Repro:
```py

import torch
from torchcodec.decoders import VideoDecoder

cpudec = VideoDecoder("/home/nicolashug/testsrc2.mp4", device="cpu")
gpudec = VideoDecoder("/home/nicolashug/testsrc2.mp4", device="cuda")

cpu_frames = cpudec[:]
gpu_frames = gpudec[:].cpu()


from torchvision.transforms.v2.functional import to_pil_image
from torchvision.utils import make_grid


for i, (cpu, gpu) in enumerate(zip(cpu_frames, gpu_frames)):
    img = to_pil_image(make_grid([cpu, gpu], nrow=2))
    img.save(f"/home/nicolashug/imgs/{i}.jpg")

torch.testing.assert_close(cpu_frames, gpu_frames.cpu())
```

We're getting pretty high atol differences:

```
AssertionError: Tensor-likes are not equal!

Mismatched elements: 165091589 / 186624000 (88.5%)
Greatest absolute difference: 48 at index (6, 0, 189, 236)
Greatest relative difference: inf at index (0, 2, 8, 4)
```

Looking at the decoded frames, they look very similar but the colors are a bit different (basically: it's a color issue, not an indexing or seeking issue). Example frames (cpu left, gpu right):

![4](https://github.com/user-attachments/assets/cea98f37-564c-4231-bb5b-027ce2ad68b3)
![27](https://github.com/user-attachments/assets/89fbe2c9-a2e2-499d-81a4-c4e42a3ead03)

Note that it's unlikely to be related or caused by https://github.com/pytorch/torchcodec/pull/413, because I'm observing the exact same differences when reverting that commit.

Thanks again @pjs102793 for the initial report. Do you have any intuition on what might be causing those differences on this specific video? From what I understand, you didn't notice major difference either on other videos, right?

**Comment by pjs102793** [comment]
Yes, the videos I have, encoded in H.264 with YUV420p, don’t seem to have any issues. I even tested with the Big Buck Bunny video, and it worked fine. The issue appears to occur only with testsrc2. One possible reason could be that testsrc2 has its colorspace information marked as unknown, which might be causing the problem.

Additionally, when I compared the values between torchaudio on GPU and torchcodec on CPU, the differences were minimal. However, the difference between torchaudio on GPU and torchcodec on GPU was comparable to the difference between torchcodec on CPU and GPU. This suggests that under certain conditions, torchcodec on GPU may produce slightly different colors.

**Comment by NicolasHug** [comment]
Pretty sure we fixed that with https://github.com/meta-pytorch/torchcodec/pull/791 and the other color-space PRs for GPU
